src/app/shared/app-flyer/app-flyer.service.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { TestBed, waitForAsync } from '@angular/core/testing';
+import { Platform } from '@ionic/angular';
+import { SharedTestingModule } from '../shared-testing.module';
+import { AppFlyerService } from './app-flyer.service';
+
+describe('AppFlyerService', () => {
+  let service: AppFlyerService;
+  let platformIs: (platformName: string) => boolean;
+  let platformReadySpy: Promise<void>;
+  let platformSpy: Platform;
+
+  beforeEach(
+    waitForAsync(async () => {
```

Comment:
Currently this test fails with error

```
  Uncaught Error: Uncaught (in promise): TypeError: this.platform.is is not a function
  TypeError: this.platform.is is not a function
```

I found this stackoverflow [answer](https://stackoverflow.com/questions/66146320/ionic-typeerror-this-platform-is-is-not-a-function-when-running-unit-tests) and also saw similar code in [app.componen.spec.ts](https://github.com/numbersprotocol/capture-lite/blob/c485e6d662cc46490cf92a7d25ff5fe3b15fd1aa/src/app/app.component.spec.ts#L16) but after 1-2 hours of trying I couldn't fix this test. 

Need help to fix it from seniors 🥺.

Reply:
I did not try the code yet, and here are some thoughts

1. These materials might be helpful, but you likely have read them

https://jasmine.github.io/api/edge/jasmine.html#.createSpyObj

https://luardo.medium.com/testing-with-mocks-spies-using-createspyobj-in-angular-1e31bf7f6147

2. `platformReadySpy` is Object, and `platformIs` is function.